src/data/tracks.json

```
@@ -81,7 +81,8 @@
   {
     "slug": "delphi",
     "name": "Delphi Pascal",
-    "core_enabled": true
+    "core_enabled": true,
+    "versioning": "exercises/{ExerciseSlug}/*est?.pas"

```

Isn't a patter like `exercises/{ExerciseSlug}/*_[T|t]est?.pas$`. going to be safer here?

---

Ah, these are not grep / regex patterns. But the correct, and strict pattern would be:

```suggestion
    "versioning": "exercises/{slug}/u{ExerciseSlug}Tests.pas"
```

- `exercises/{ExerciseSlug}` will look for `exercises/BookStore/` folder instead of `exercises/book-store`
- I don't support `?` or `*`, but I agree. If this were a Regex, what you propose _is_ safer!

If you can confirm:

- The test file always starts with a lowercase `u`
- The test file name then has `CamelCase` for the slug
- The test file name then has `Tests.pas`

In this case, my suggested pattern is correct.

---

@rpottsoh, sorry, this was actually intended for you 🗡 

> Furthermore the test suite always contains the word test in the its filename, though sometimes it may be Test or Tests.

Is there a reason it's sometimes `Test` or `Tests` (read: is it legacy??), because if it's all converging to `Tests`, I would suggest only supporting one of the two. 

> Windows is not case sensitive on filenames.

Unfortunately Github is

> Hopefully wildcards are acceptable.

Wildcards currently don't work, but for you I can add them. Let me know the details and I'll make it work.

---

> Is there a reason it's sometimes Test or Tests (read: is it legacy??), because if it's all converging to Tests, I would suggest only supporting one of the two.

No reason.  I got smarter as I added exercises over time and there was no impetus to be so rigid in naming as it satisfies the default test pattern.  This track just turned 3 on the 3rd.

---

> The test file always starts with a lowercase u

Yes, all test suites start with `u`.
> The test file name then has CamelCase for the slug

Yes.  There are no hyphens in any of the file names.

> The test file name then has Tests.pas

Not exactly.  All test suites contain either `Test` or `test` at some position in the file name.  Over time I became more consistent at putting Test or Tests towards the end of the file name.

---

Thanks @SleeplessByte was aware that it wasn't regular expression but also wasn't sure how to express the idea.

---

Alright. May I suggest that we _over time_ normalise the `test` file name? The _reason_ we don't allow for wildcards is that I am trying to keep the entire application API call free. Currently I need just a single one to do the `unimplemented` thing. If _most_ exercises end with `Test` or `Tests`, I can add a special case for you so that we look for _either_?

Why is it working currently? Well, I try to load the actual file from `raw.githubusercontent.com`!  Using wildcards will force me to "fetch the tree for the repo, then match the files against the pattern, then fetch the file", the first requiring an API call.